relay-java/src/main/java/com/genymobile/gnirehtet/Main.java

```diff
@@ -50,7 +51,12 @@ String getDescription() {
             @Override
             void execute(CommandLineArguments args) throws Exception {
                 Log.i(TAG, "Installing gnirehtet client...");
-                execAdb(args.getSerial(), "install", "-r", "gnirehtet.apk");
+                File apk = new File("/opt/gnirehtet/gnirehtet.apk");
```

Comment:
A new argument (--apk ?) should be added with the install, reinstall and run args.
I am not very familiar with cargo, so is it possible to have something like "configure" script (usually found in make based projects) ?
If yes, then it's possible to have pkgbuild compiled binary to default to distribution specific location defined in pkgbuild for lookup while its the current directory otherwise.
And, explicitly specifying --apk overrides all logic.
What do you say?

Reply:
> A new argument (--apk ?) should be added with the install, reinstall and run args.

`gnirehtet install` is basically a shortcut for:

    adb install gnirehtet.apk

Passing the `--apk` as a parameter would defeat its purpose. I prefer an env variable for that.

> I am not very familiar with cargo, so is it possible to have something like "configure" script

Either we "configure" the variable at compile time (to hardcode its path in the resulting binary) in both cargo and gradle, either we retrieve its value at runtime in the environment variable (as I suggested [here](https://github.com/Genymobile/gnirehtet/pull/59#discussion_r151865005)).

I'm ok for both, but currently I prefer the latter, since it offers more flexibility. Do you prefer the former, and why?

For now, I don't know how to "configure" in cargo, but that's probably possible to get something similar…